### DISCUSSION TITLE
Switching from scheduled playlist to other after the playlist is finished.

### DISCUSSION THREAD

**Discussion by hjkaddict** [savonet/liquidsoap#2074] [Q&A] [answered]
Hi, liquidsoap developers and people,

I am building up small radio station. But I just cannot solve this simple problem. Please understand, I am not from programming background:) 

So I have this playlist 'rotation' and 'test'. In 'rotation' playlist, I have tons of tracks in there and playback randomly. On the other hand, 'test' playlist had only 12 tracks with mode=normal. What I would like to do is to play the 'test' once in specific time and switch to the 'rotation' playlist after the tracks in 'test' are cycled one time.

```
rotation = playlist(mode="randomize", "/radiosource", reload_mode="rounds", reload=1)
test = playlist(mode="normal", "/testsource")
```

I assume that if I want to switch between two playlists, I have to do use switch function as below:

`
radio = switch([({ 23h }, test), ({ true }, rotation)])
`

obviously it works in half because it starts to play the 'test' on 23h but when the time comes (24h) it switch back to 'rotation' again. Like I said, what I would like to do is to switch back to 'rotation' again after all tracks in 'test' are played. 

Sorry for asking such a simple question. Any small hints or simple keyword would be really appreciated!

**Comment by smimram** [comment, marked as answer]
I would change the logic and use a queue, here and push all the files at 23h. Something like:
```
rotation = ...
test = request.queue()
radio = fallback([test, rotation])
thread.when({23h},  {list.iter(fun(f) -> test.push.uri(f), playlist.files("/testsource"))})
```

**Reply by hjkaddict** [reply]
First the code kept complaining but I forgot to update to v2.0. Once I have done, it works. I should look more into documentation:) Thanks a lot. Fanstastic features!
